src/blinding.cpp

```diff
@@ -298,6 +301,37 @@ std::pair<uc32, cleared_uc32> blind25_key_pair(
     return result;
 }
 
+static const auto version_blinding_hash_key_sig = to_unsigned_sv("VersionCheckKey_sig"sv);
+
+std::pair<uc32, cleared_uc64> blind_version_key_pair(ustring_view ed25519_sk) {
+    std::array<unsigned char, 64> ed_sk_tmp;
+    if (ed25519_sk.size() == 32) {
+        std::array<unsigned char, 32> pk_ignore;
+        crypto_sign_ed25519_seed_keypair(pk_ignore.data(), ed_sk_tmp.data(), ed25519_sk.data());
+        ed25519_sk = {ed_sk_tmp.data(), 64};
+    }
+    if (ed25519_sk.size() != 64)
+        throw std::invalid_argument{
+                "blind_version_key_pair: Invalid ed25519_sk is not the expected 32- or 64-byte "
+                "value"};
+
+    uc32 seed;
```

Comment:
This ought to be in a cleared_uc32, but actually we can optimize that a bit by moving the `cleared_uc64 sk;` from further down up here to do double-duty (and avoid some secure erases):

```C++
    std::pair<uc32, cleared_uc64> result;
    auto& [pk, sk] = result;

    crypto_generichash_blake2b(
            sk.data(),
            32,
            ed25519_sk.data(),
            /* ... */);
    
    // Reuse `sk` to avoid needing extra secure erasing:
    if (0 != crypto_sign_ed25519_seed_keypair(pk.data(), sk.data(), sk.data()))
```

Reply:
Nice - have updated it

Reply:
So I've included this change but just wanted to flag that it ended up resulting in changing the output of the function (see unit test changes below), I don't really understand _why_ the result would have changed though because it seems to only use the first 32 bytes of `sk` 😕 

```

~/LibSession-Util/tests/test_blinding.cpp:282: FAILED:
  CHECK( oxenc::to_hex(seckey.begin(), seckey.end()) == "91faddc2c36da4f7bcf24fd977d9ca5346ae7489cfd43c58cad9eaaa6ed60f6988e8adb27e7b8ce776fcc25b" "c1501fb2888fcac0308e52fb10044f789ae1a8fa" )
with expansion:
  "4091bdaafefd7ddc3398db877c894716b2b24f12dac15ad414a3e4f0b6ac1c6788e8adb27e7-
  b8ce776fcc25bc1501fb2888fcac0308e52fb10044f789ae1a8fa"
  ==
  "91faddc2c36da4f7bcf24fd977d9ca5346ae7489cfd43c58cad9eaaa6ed60f6988e8adb27e7-
  b8ce776fcc25bc1501fb2888fcac0308e52fb10044f789ae1a8fa"

~/LibSession-Util/tests/test_blinding.cpp:291: FAILED:
  CHECK( oxenc::to_hex(signature.begin(), signature.end()) == "143c2c9828f7680ee81e6247bc7aa4777c4991add87cd724149b00452bed4e920fa57daf4627c68f43fcbddb" "2d465d5ea11def523f3befb2bbee39c769676305" )
with expansion:
  "adfb25eafec6fb0037fc39145e64badfbdd08cc95a77c01577ecc623fd856fb05b90a921fc6-
  b805e6a730ca9505e1c069f256020a76beb2ecbb6e32c47e12104"
  ==
  "143c2c9828f7680ee81e6247bc7aa4777c4991add87cd724149b00452bed4e920fa57daf462-
  7c68f43fcbddb2d465d5ea11def523f3befb2bbee39c769676305"

```

Reply:
That seems like a problem.  I'll investigate.

Reply:
It seems my suggestion was wrong: libsodium does not allow the seed input and secret key output to overlap here.  I've pushed a fix to put the temp seed in a separate cleared_uc32 instead.

Reply:
Ah no worries, thanks 🙏